src/main/java/graphql/schema/idl/TypeDefinitionRegistry.java

```diff
@@ -490,43 +492,138 @@ public Map<String, DirectiveDefinition> getDirectiveDefinitions() {
         return new LinkedHashMap<>(directiveDefinitions);
     }
 
+    /**
+     * Returns true if the registry has a type of the specified {@link TypeName}
+     *
+     * @param typeName the type name to check
+     *
+     * @return true if the registry has a type by that type name
+     */
     public boolean hasType(TypeName typeName) {
         String name = typeName.getName();
+        return hasType(name);
+    }
+
+    /**
+     * Returns true if the registry has a type of the specified name
+     *
+     * @param name the name to check
+     *
+     * @return true if the registry has a type by that name
+     */
+    public boolean hasType(String name) {
         return types.containsKey(name) || ScalarInfo.GRAPHQL_SPECIFICATION_SCALARS_DEFINITIONS.containsKey(name) || scalarTypes.containsKey(name) || objectTypeExtensions.containsKey(name);
     }
 
+    /**
+     * Returns am optional {@link TypeDefinition} of the specified type or {@link Optional#empty()}
+     *
+     * @param type the type to check
+     *
+     * @return an optional {@link TypeDefinition} or empty if it's not found
+     */
     public Optional<TypeDefinition> getType(Type type) {
-        String typeName = TypeInfo.typeInfo(type).getName();
-        return getType(typeName);
+        return getType(typeName(type));
     }
 
+    /**
+     * Returns am optional {@link TypeDefinition} of the specified type with the specified class or {@link Optional#empty()}
+     *
+     * @param type   the type to check
+     * @param ofType the class of {@link TypeDefinition}
+     *
+     * @return an optional {@link TypeDefinition} or empty if it's not found
+     */
     public <T extends TypeDefinition> Optional<T> getType(Type type, Class<T> ofType) {
-        String typeName = TypeInfo.typeInfo(type).getName();
-        return getType(typeName, ofType);
+        return getType(typeName(type), ofType);
     }
 
+    /**
+     * Returns am optional {@link TypeDefinition} of the specified type name or {@link Optional#empty()}
+     *
+     * @param typeName the type to check
+     *
+     * @return an optional {@link TypeDefinition} or empty if it's not found
+     */
     public Optional<TypeDefinition> getType(String typeName) {
+        return Optional.ofNullable(getTypeOrNull(typeName));
+    }
+
+    /**
+     * Returns am optional {@link TypeDefinition} of the specified type name with the specified class or {@link Optional#empty()}
+     *
+     * @param typeName the type to check
+     * @param ofType   the class of {@link TypeDefinition}
+     *
+     * @return an optional {@link TypeDefinition} or empty if it's not found
+     */
+    public <T extends TypeDefinition> Optional<T> getType(String typeName, Class<T> ofType) {
```

Comment:
Just return `null` instead using Optional? Saves also an object allocation

Reply:
Ah .. I see you have it actually below ... how about just killing this method then?